### PR TITLE
Fix/nex 253/fix cache management

### PR DESCRIPTION
**Pull request by boajer**
Fixed:
 - cache key retrieval
 - DeliverTenant object serialisation/deserialisation 
 - helper script

### CODE REVIEW

File: model/api/TenantManagement.php [hunk 1]
```diff
@@ -39,7 +40,7 @@ abstract class TenantManagement extends ConfigurableService implements TenantMan
     protected $collection;
 
     /**
-     * @var common_persistence_Persistence
+     * @var common_persistence_KvDriver
```
Comment (julien-sebire) [inline model/api/TenantManagement.php]: Do we necessarily have to constraint to one sort of persistence? The `getPersistence` method below returns the current configured persistence though...

File: model/api/TenantManagement.php [hunk 2]
```diff
@@ -79,26 +80,34 @@ public function getTenantById($id)
     }
 
     /**
-     * @param $key
-     * @return mixed
+     * @param string $key
+     * @return null|[]|DeliverTenant[]
```
Comment (julien-sebire) [inline model/api/TenantManagement.php]: null in return type gives me the idea that we could accept that json_decode returns a null, meaning that the key doesn't exist. Even if it's clever, Isn't it clearer to test that it exists instead?

Comment (boajer) [reply model/api/TenantManagement.php]: Do you mean the call to cache persistence to like an `::exists($key)` I afraid we can land up with a doubling calls to the storage, that is not good from performance perspective

File: model/api/TenantManagement.php (hunk 1 above)
Comment (boajer) [reply model/api/TenantManagement.php]: We expect KV persistence only there, so I've added extra validation

File: model/api/TenantManagement.php (hunk 2 above)
Comment (julien-sebire) [reply model/api/TenantManagement.php]: Fair enough. Maybe just add a comment with this in mind to explain the choice here to get along with a null from json_decode... Opinion?

File: model/api/TenantManagement.php (hunk 1 above)
Comment (julien-sebire) [reply model/api/TenantManagement.php]: ok for me.

**Pull request merged**